src/common/HostInterface.c

```diff
@@ -109,21 +109,21 @@ void AddDataToDataToSend(UCHAR * bytNewData, int Len)
 
 	SetLED(TRAFFICLED, TRUE);
 
-	sprintf(HostCmd, "BUFFER %d", bytDataToSendLength);
+	snprintf(HostCmd, sizeof(HostCmd), "BUFFER %d", bytDataToSendLength);
 	QueueCommandToHost(HostCmd);
 }
 
 char strFault[100] = "";
 
-VOID DoTrueFalseCmd(char * strCMD, char * ptrParams, BOOL * Value)
+bool DoTrueFalseCmd(char * strCMD, char * ptrParams, BOOL * Value)
 {
 	char cmdReply[128];
 
 	if (ptrParams == NULL)
 	{
-		sprintf(cmdReply, "%s %s", strCMD, (*Value) ? "TRUE": "FALSE");
+		snprintf(cmdReply, sizeof(cmdReply), "%s %s", strCMD, (*Value) ? "TRUE" : "FALSE");
 		SendReplyToHost(cmdReply);
-		return;
+		return false;
```

Comment:
While you didn't document what the return value of DoTrueFalseCmd() is intended to indicate, I think you meant for false to indicate an Error.  If that is your intent, it should return True when ptrParams == NULL.

Reply:
That's not the intent of the new return value. The return value merely indicates if the client has provided a command which updates `*Value`. I added some function documentation for this.